microsoft-azure-storage/src/com/microsoft/azure/storage/core/Utility.java

```diff
@@ -116,13 +130,22 @@
      * Used to create Json parsers and generators.
      */
     private static final JsonFactory jsonFactory = new JsonFactory();
-    
+
     /**
-     * A factory to create SAXParser instances.
+     * Thread local for SAXParser.
      */
-    private static final ThreadLocal<SAXParserFactory> saxParserFactory = new ThreadLocal<SAXParserFactory>() {
-        @Override public SAXParserFactory initialValue() {
-            return SAXParserFactory.newInstance();
+    private static final ThreadLocal<SAXParser> saxParserThreadLocal = new ThreadLocal<SAXParser>() {
+        SAXParserFactory factory;
+        @Override public SAXParser initialValue() {
+            factory = SAXParserFactory.newInstance();
+            factory.setNamespaceAware(true);
+            try {
+                return factory.newSAXParser();
+            } catch (SAXException e) {
+                throw new RuntimeException("Unable to create SAXParser", e);
+            } catch (ParserConfigurationException e) {
+                throw new RuntimeException("Check parser configuration", e);
+            }
```

Comment:
Did I misunderstand your comment?. It would not be possible to throw checked exceptions in initalValue() and hence the RuntimeExceptions.